mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,122 @@
-import React from 'react';
+import { type IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import {
+    type FontAwesomeIconStyle,
+    type Props,
+} from '@fortawesome/react-native-fontawesome';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import React, { useCallback, useState } from 'react';
 import { Pressable } from 'react-native';
+import {
+    type PressableProps,
+    type StyleProp,
+    type ViewStyle,
+} from 'react-native';
 
 import { Text } from '~/bundles/common/components/components';
 
 import { styles } from './styles';
 
 type Properties = {
-    label: string;
-    onPress: () => void;
-};
+    label?: string;
+    isPrimary?: boolean;
+    labelStyles?: StyleProp<ViewStyle>;
+    icon?: IconDefinition;
+    iconProps?: Omit<Props, 'icon'>;
+} & PressableProps;
+
+type StyleRecord = Record<string, unknown>;
+
+const Button: React.FC<Properties> = ({
+    label = '',
+    labelStyles,
+    style,
+    isPrimary = true,
```

Comment:
Let's create an enum for button state, e.x:
```
const ButtonStatus = {
    FILLED: 'Filled',
    OUTLINE: 'Outline',
    GHOST: 'Ghost',
} as const;
```
Then in the switch case, you return button styles depending on the button type, the default value can be ButtonStatus.FILLED

Reply:
fix